app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -403,6 +414,32 @@ class WireActivityViewModel @Inject constructor(
         globalAppState = globalAppState.copy(updateAppDialog = false)
     }
 
+    fun observePersistentConnectionStatus() {
+        viewModelScope.launch {
+            coreLogic.getGlobalScope().observePersistentWebSocketConnectionStatus().let { result ->
+                when (result) {
+                    is ObservePersistentWebSocketConnectionStatusUseCase.Result.Failure -> {
+                        appLogger.e("Failure while fetching persistent web socket status flow from wire activity")
+                    }
+
+                    is ObservePersistentWebSocketConnectionStatusUseCase.Result.Success -> {
+                        result.persistentWebSocketStatusListFlow.collect { statuses ->
+
+                            if (statuses.any { it.isPersistentWebSocketEnabled }) {
+                                if (!servicesManager.isPersistentWebSocketServiceRunning()) {
+                                    servicesManager.startPersistentWebSocketService()
+                                }
+                            } else {
+                                servicesManager.stopPersistentWebSocketService()
```

Comment:
this stop can cause an issue if a user start and stop the web socket the app will be stuck on connecting (if the app is in the foreground)

Reply:
let's tackle it in a different fix 👍🏼